Core/Scripts: implement warrior spell Storm Bolt

HannibalRoG opened the pull request:
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Implement Warrior spell Storm Bolt (stun).

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
Compiled and tested in game.

HannibalRoG committed:
Implement spell Warrior Stormbolt.

HannibalRoG committed:
Sql for Stormbolt.

Temppus commented on src/server/scripts/Spells/spell_warrior.cpp:
```diff
@@ -683,6 +685,45 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
+            Unit* caster = GetCaster();
+
+            if (Unit* target = GetHitUnit())
+                caster->CastSpell(target, SPELL_WARRIOR_STORM_BOLT_STUN, true);
```
Dont need to store caster pointer. Just GetCaster()->CastSpell(.....);

HannibalRoG replied:
Ups, I passed this comment, thanks @tkrolki.
We fix it soon.

DoctorKraft commented on src/server/scripts/Spells/spell_warrior.cpp:
```diff
@@ -53,6 +53,8 @@ enum WarriorSpells
     SPELL_WARRIOR_SECOUND_WIND_TRIGGER_RANK_2       = 29842,
     SPELL_WARRIOR_SHIELD_SLAM                       = 23922,
     SPELL_WARRIOR_SLAM                              = 50782,
+    SPELL_WARRIOR_STORM_BOLT                        = 107570,
```
unused

DoctorKraft commented on src/server/scripts/Spells/spell_warrior.cpp:
```diff
@@ -683,6 +685,45 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_STORM_BOLT,
```
same unused

Shauren replied:
Don't validate the spell you are attaching the script to (107570), it is done by ObjectMgr already and the script is never created

DoctorKraft commented:
Wrong effect also, no ? You need add this in `SPELL_EFFECT_DUMMY` (Effect1) with `OnEffectHitTarget`

P-Kito commented:
SPELL_EFFECT_DUMMY -> OnEffectHitTarget -> Cast 132169

Make sure that the stormbolt flys to the target before stunning it (small delay if it flys a long range)

HannibalRoG commented:
Now when the day of love passes we review and fix it, by the way, congratulations to all by February 14th.

Ascathor commented:
@HannibalRoG Same right back to you, even if it's not a celebration to me. However, it is well hoped that you had a fine day and enjoyed it with your loved one.

HannibalRoG committed:
Fixes for Strombolt Warrior.

HannibalRoG commented:
Any comments now?

Commit:
Update and rename RoG_2017_02_13_world.sql to 2017_02_29_19130_world.sql

- removed unneeded blank first line
- removed ' ' from integer value (TC SQL code style)
- changed file name according to Aokromes' recommendation

tkrokli commented:
Maybe you could reply to Temppus in his comment https://github.com/TrinityCore/TrinityCore/pull/19130#pullrequestreview-21626837. I don't know if his suggestion is correct or not.

HannibalRoG committed:
More fix, thanks Temppus.

HannibalRoG committed:
Merge branch 'Spell_Warrior_Stormbolt' of https://github.com/Edreisantafe/TrinityCore into Spell_Warrior_Stormbolt

tkrokli commented on src/server/scripts/Spells/spell_warrior.cpp:
```diff
@@ -683,6 +684,43 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
+
+        void HandleOnHit(SpellEffIndex /*effIndex*/)
+        {
+            
```
White space / unneeded blank line

HannibalRoG committed:
White space.

tkrokli commented:
@HannibalRoG : Looks OK so far, all we need now is for one of the main devs to verify/validate this PR and merge it.

wonder2011 commented:
What about this PR?
